source/development/testing.rst

```diff
@@ -1,6 +1,21 @@
 Testing
 =======
-
+The primary goal of the Bao Test Framework is to optimize the testing process
+of Bao. It encompasses three core components: (i) a C code library designed for
+real-time test handling, (ii) a build system based on Nix, and (iii) a Python
+API for comprehensive test management.
+
+1. **C Code Library** - This library comprises a set of utilities that simplify
```

Comment:
```suggestion
1. **C Library** - This library comprises a set of utilities that simplify
```

C library is enough, "C code" is kind of a pleonasm.

Reply:
Agree, integrated.